src/components/FormStep/index.js

```diff
@@ -834,6 +834,9 @@ const FormStep = ({
         isOpen={isFormSaveModalOpen}
         closeModal={closeFormStepSaveModal}
         onSaveConfirm={onSaveConfirm}
+        onSessionDestroyed={() => {
+          localStorage.removeItem(form.uuid);
+        }}
```

Comment:
we should not use the global localStorage here, but instead use the `removeSubmissionId` callback from the `useRecycleSubmission` hook (which uses `useLocalStorage` under the hood).